### PR TITLE
Added tests for ImmutableList and some other minor stuff.

### PR DESCRIPTION
**Pull request by vontman**

### CODE REVIEW

File: src/main/java/edu/immutables/list/impl/Cons.java [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+    @Override
+    public ImmutableList<T> dropWhile(Predicate<? super T> p) {
+        return reverse(reverse(this).takeWhile(p.negate()));
+    }
+
+    @Override
+    public ImmutableList<T> take(int length) {
+        if (length == 0)
+            return new Empty<> ();
+        return new Cons<>(head, tail.take(length - 1));
```
Comment (tknawara) [inline src/main/java/edu/immutables/list/impl/Cons.java:89]: what if length is greater than the length of the original list, should you throw `IndexOutOfBoundException` or take the hold list ?

File: src/main/java/edu/immutables/list/impl/Cons.java [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+    }
+
+    @Override
+    public ImmutableList<T> drop(int length) {
+        return null;
+    }
```
Comment (tknawara) [inline src/main/java/edu/immutables/list/impl/Cons.java:95]: Not yet implemented ?

File: src/main/java/edu/immutables/list/impl/Cons.java [hunk 3]
```diff
@@ -0,0 +1,160 @@
...
+
+    @Override
+    public ImmutableList<T> append(ImmutableList<T> t) {
+        if (size == 1)
+            return new Cons<>(head, t);
+        return new Cons<>(head, tail.append(t));
+    }
```
Comment (tknawara) [inline src/main/java/edu/immutables/list/impl/Cons.java]: you don't have the if condition, simply return `new Cons<>(head, tail.append())` append implementation in `Empty` will end the recursive calls.

File: src/test/java/edu/immutables/list/ListTest.groovy [hunk 4]
```diff
@@ -0,0 +1,171 @@
...
+
+        where:
+        list                                         || size
+        emptyList()                                  || 0
+        emptyList().addHead(1)                       || 1
+        emptyList().addHead(1).addHead(2)            || 2
+        emptyList().addHead(1).addHead(2).addHead(3) || 3
+    }
+
+    def "Test append operation"() {
```
Comment (tknawara) [inline src/test/java/edu/immutables/list/ListTest.groovy:24]: wal3a 👍

**Review by tknawara — COMMENTED** [review]
Nice work, very good testing

File: src/main/java/edu/immutables/list/impl/Cons.java (hunk 1 above)
Comment (vontman) [reply src/main/java/edu/immutables/list/impl/Cons.java:89]: In `Empty` class
```java
@Override
    public ImmutableList<T> take(int length) {
        if (length != 0)
            throw new IndexOutOfBoundsException();
        return this;
    }
```

File: src/main/java/edu/immutables/list/impl/Cons.java (hunk 2 above)
Comment (vontman) [reply src/main/java/edu/immutables/list/impl/Cons.java:95]: Ops !

**Pull request merged**